packages/pieces/community/google-slides/src/lib/actions/generate-from-template.ts

```diff
@@ -0,0 +1,120 @@
+            props: async ({auth, template_presentation_id}) => {
+                if (!template_presentation_id || !auth)
+                    return {};
+        
+                const presentation = await getSlide(auth["access_token"] as unknown as string, template_presentation_id as unknown as string);
+                if (!presentation)
+                    return {}
+
+                const fields = {
+                    title: Property.ShortText({
+                        displayName: 'Presentation Title',
+                        description: 'Title of the new presentation',
+                        defaultValue: `Copy of: ${presentation.title}`,
+                        required: true,
+                    })
+                } as DynamicPropsValue;
+        
+                presentation.slides?.forEach(slide => {
+                    slide.pageElements?.forEach((element: PageElement) => {
+                        if (element.shape?.text?.textElements) {
+                            element.shape.text.textElements.forEach(textElement => {
+                                const content = textElement?.textRun?.content
+                                if (content) {
+                                    const matches = content.match(/\{\{([^}]+)\}\}/g);
+                                    if (matches) {
+                                        matches.forEach((match: string) => {
+                                            const matchValue = match.replace(/[{}]/g, '');
+                                            const varName = matchValue.trim();
+                                            fields[matchValue] = Property.ShortText({
+                                                displayName: varName.replace(/_/g, ' ').replace(/\b\w/g, l => l.toUpperCase()),
+                                                description: `Value for "${matchValue}"`,
+                                                required: false,
+                                            });
+                                        });
+                                    }
+                                }
+                            });
+                        }
+                    });
+                });
+        
+                return fields;
+            }
+        })
+    },
+    async run(context) {
+        const { access_token } = context.auth;
+        const { template_presentation_id, table_data } = context.propsValue;
+
+        try {
+            const authClient = new OAuth2Client();
+            authClient.setCredentials(context.auth);
+            
+            const drive = google.drive({ version: 'v3', auth: authClient });
+                
+            const copyResponse = await drive.files.copy({
+                fileId: template_presentation_id as string,
+                requestBody: {
+                    name: table_data["title"] || "New Presentation"
+                },
+                supportsAllDrives: true
```

Comment:
Ah got it you hard-coded it
I'm torn on this: it's simpler for everyone, but it's not consistent with other Google pieces / actions
worst case it can be changed into a props later